pkg/watcher/certificate_manager.go

```diff
@@ -42,15 +42,26 @@ type SubjectAltName struct {
 }
 
 type CertificateManager struct {
-	kcpClient           client.Client
-	kyma                *v1beta2.Kyma
-	caCertCache         *CertificateCache
-	certificateName     string
-	secretName          string
-	istioNamespace      string
-	remoteSyncNamespace string
-	caCertName          string
-	additionalDNSNames  []string
+	kcpClient       client.Client
+	caCertCache     *CACertificateCache
+	certificateName string
+	secretName      string
+	config          CertificateConfig
+}
+
+type CertificateConfig struct {
+	// IstioNamespace represents the cluster resource namespace of istio
+	IstioNamespace string
+	// RemoteSyncNamespace indicates the sync namespace for Kyma and module catalog
+	RemoteSyncNamespace string
+	// CACertificateName indicates the Name of the CA Root Certificate in the Istio Namespace
+	CACertificateName string
+	// AdditionalDNSNames indicates the DNS Names which should be added additional to the Subject
+	// Alternative Names of each Kyma Certificate
+	AdditionalDNSNames []string
+	Duration           apimetav1.Duration
```

Comment:
Why we need to use the `apimetav1` and not `time`?

Reply:
Because it's the contract definition of this field in `certmanagerv1.CertificateSpec`

Reply:
I changed to `time.Duration`, it's more suitable for this config